### ISSUE TITLE
Diffing composer.lock files kills the host by eating all memory

### ISSUE THREAD

**Issue by jurgenhaas** [Wilfred/difftastic#373]
This tool is amazing, thanks for providing it for free.

I ran into an issue with a composer.lock file. When diffing it, all the available memory and all of the swap memory will be utilized within seconds. And we're talking about ab 64 GB of available memory.

The 2 files are attached and I was comparing them with version 0.35.0 using this command:

```
difft --language=json composer.org.lock composer.lock
```

I had to kill the process by SSH'ing into that host, because the local keyboard was quickly unavailable.

[composer.zip](https://github.com/Wilfred/difftastic/files/9564808/composer.zip)

**Comment by Wilfred** [comment]
Just to confirm, have you set `DFT_BYTE_LIMIT` or `DFT_GRAPH_LIMIT`?

**Comment by jurgenhaas** [comment]
No, I haven't. Am I supposed to do so? Can't find them in the manual.

**Comment by Wilfred** [comment]
Gotcha. They're visible in --help but not in the manual.

They control the memory limits, so you can definitely OOM if you increase them. Since you haven't set them, this is definitely a bug!

**Comment by Wilfred** [comment]
```
$ DFT_GRAPH_LIMIT=10 RUST_LOG=debug difft --language=json composer.org.lock composer.lock
 2022-09-16T03:55:57.980Z INFO  difft::options > CLI arguments: ["composer.org.lock", "composer.lock"]
 2022-09-16T03:55:58.406Z INFO  difft::diff::dijkstra > LHS nodes: 76618 (5 toplevel), RHS nodes: 77810 (5 toplevel)
```

There are two issues here:

* The graph limit isn't being respected for some reason
* The unchanged node heuristics aren't doing a very good job, so we're trying to diff huge inputs

**Comment by Wilfred** [comment]
Memory usage gets very high around 2,200,700 graph nodes.
